python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py

```diff
@@ -36,6 +54,34 @@ def build_components_from_decl_node(
     raise NotImplementedError(f"Unknown component type {decl_node}")
 
 
+def component_type_from_yaml_decl(
+    context: ComponentLoadContext, decl_node: YamlComponentDecl
+) -> Type:
+    parsed_defs = decl_node.defs_file_model
+    if parsed_defs.component_type.startswith("."):
+        component_registry_key = parsed_defs.component_type[1:]
+
+        # Iterate over Python files in the folder
+        for py_file in decl_node.path.glob("*.py"):
+            module_name = py_file.stem
+
+            module = load_module_from_path(module_name, decl_node.path / f"{module_name}.py")
+
+            for _name, obj in inspect.getmembers(module, inspect.isclass):
+                assert isinstance(obj, Type)
+                if (
+                    is_registered_component(obj)
+                    and component_registry_key_of(obj) == component_registry_key
+                ):
+                    return obj
```

Comment:
Perhaps best left for a follow-up but I am partial to avoiding `load_module_from_path` and instead requiring the containing code location to be installed in the python environment

Reply:
Yeah let's do that in follow up